include/deal.II/base/exceptions.h

```diff
@@ -1228,16 +1228,19 @@ namespace StandardExceptions
                                            dealii::ExcCudaError(cudaGetErrorString(error_code)))
 #endif
 
-// A static assert which checks a compile-time condition. This is nothing more
-// than a wrapper around the C++11 <code>static_assert</code> declaration, which
-// we need to disable for non-C++11.
 
+// A static assert which checks a compile-time condition. This is nothing more
+// than a wrapper around the C++11 <code>static_assert</code> declaration,
+// which we need to disable for non-C++11.
+//
 // Just like <code>static_assert</code>, it takes two arguments, a
 // <code>constexpr</code> condition and an error message.
 #if defined(DEAL_II_WITH_CXX11)
 #define DEAL_II_STATIC_ASSERT(...) static_assert(__VA_ARGS__)
 #else
-#define DEAL_II_STATIC_ASSERT(...) /* do nothing */
+#define DEAL_II_STATIC_ASSERT_DUMMY(L) typedef char __static_assert__##L
+#define DEAL_II_STATIC_ASSERT_DUMMY2(L) DEAL_II_STATIC_ASSERT_DUMMY(L)
+#define DEAL_II_STATIC_ASSERT(...) DEAL_II_STATIC_ASSERT_DUMMY2(__LINE__)
```

Comment:
Very clever. Why do you need the second indirection with `DEAL_II_STATIC_ASSERT_DUMMY2`? In fact, why do you need the first indirection even?